File: Data Structures & Algorithms/combination-target-sum/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @returns {number[][]}
     */
    combinationSum(nums, target) {
        let length=nums.length,ans=[];
        const dfs=(arr,sum,index)=>{
        if(sum==target){
            ans.push([...arr]);
            return;
        }
        if(sum>target || index>=length) return;
        //just move forward with the current number till their sum greater then target
        arr.push(nums[index]);
        dfs(arr,sum+nums[index],index);

        //remove last inserted value
        arr.pop();
        dfs(arr,sum,index+1);
        }
        dfs([],0,0);
        return ans;
    }
}

File: Data Structures & Algorithms/move-zeroes/submission-1.java
class Solution {
    public void moveZeroes(int[] nums) {
        int length=nums.length;
        int nonzero_index = 0;
        for(int i=0;i<length;i++){
            if(nums[i]!=0){
                int temp = nums[nonzero_index];
                nums[nonzero_index]=nums[i];
                nums[i]=temp;
                nonzero_index++;
            }
        }
    }
}